gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/jvmkill.c

```diff
@@ -17,9 +17,18 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-
+#include <stdlib.h>
 #include <jvmti.h>
 
+void logMessage(char * message){
+  FILE * logFile = fopen("jvmkill.log", "w");
+  if(logFile == NULL){
+    fprintf(stderr, "Error opening log file");
+  }
+  fputs(message, logFile);
+  fclose(logFile);
```

Comment:
Since you are closing the file in every call, so it should be opened in append mode just in case the code has to dump more than one line. Better option will be to not close and have it initialized on first call (for example having logFile as a local static variable is easiest option when thread-safety is not a concern like here).

Also the name of file can be a bit unique for cases where this may have happened more than once over the lifetime of a server. So appending a random value to jvmkill.log like number of seconds returned by time(), or total microseconds using gettimeofday will be better in my opinion.

Reply:
keeping the name of the file same("jvmkill.log") and opening it in append mode so that if log is written multiple times, we can see it in one place. log gets created in the work directory . Changes in new commit